components/datetime/src/time_zone.rs

```diff
@@ -745,59 +853,3 @@ impl FormatTimeZone for Bcp47IdFormat {
         Ok(Ok(()))
     }
 }
-
-#[derive(PartialEq)]
-enum ZeroPadding {
-    /// Add zero-padding.
-    On,
-
-    /// Do not add zero-padding.
-    Off,
-}
-
-/// Formats a time segment with optional zero-padding.
-fn format_time_segment<W: writeable::PartsWrite + ?Sized>(
-    sink: &mut W,
-    n: u8,
-    padding: ZeroPadding,
-) -> fmt::Result {
-    debug_assert!((0..60).contains(&n));
-    if padding == ZeroPadding::On && n < 10 {
-        sink.write_char('0')?;
-    }
```

Comment:
Praise: Thanks for using the FixedDecimalFormatter here. I didn't realize we still had code that wasn't using it.